src/lib.rs

```diff
@@ -316,7 +318,53 @@ pub fn select_coin(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let inputs_srd = inputs.to_vec();
+    let inputs_fifo = inputs.to_vec();
+    let inputs_ll = inputs.to_vec();
+
+    let best_result = Arc::new(Mutex::new(None::<SelectionOutput>));
+
+    let best_result_srd = Arc::clone(&best_result);
+    let best_result_fifo = Arc::clone(&best_result);
+    let best_result_ll = Arc::clone(&best_result);
+
+    let options_srd = options.clone();
+    let options_fifo = options.clone();
+    let options_ll = options.clone();
+
+    let srd_handle = thread::spawn(move || {
+        if let Ok(result) = select_coin_srd(&inputs_srd, options_srd) {
+            let mut best_result = best_result_srd.lock().unwrap();
+            if best_result.is_none() || result.waste.0 < best_result.as_ref().unwrap().waste.0 {
```

Comment:
Put all the functions in a `Vec` and iterate over them,
In the process spawn the thread and mutate the mutex if a better solution is found.
This avoid boilerplate code being repeated.